vm/qemu/qemu.go

```diff
@@ -529,13 +531,42 @@ func (inst *instance) boot() error {
 	return nil
 }
 
+// "vfio-pci,host=BN:DN.{{FN%8}},addr=0x11".
+func handleVfioPciArg(arg string, index int) string {
+	if index > 7 {
+		args1 := strings.Split(arg, "vfio-pci,host=")
+		if len(args1) != 2 || args1[1] == "" {
+			return arg
+		}
+		args2 := strings.Split(args1[1], ":")
+		if len(args2) != 2 || args2[0] == "" || args2[1] == "" {
+			return arg
+		}
+		args3 := strings.Split(args2[1], ".")
+		if len(args3) != 2 || args3[0] == "" || args3[1] == "" {
+			return arg
+		}
+		i := new(big.Int)
+		i.SetString(args3[0], 16)
+		devno := i.Int64() + int64(index/8)
+		str := ":" + args3[0]
+		fix := fmt.Sprintf(":%02x", devno)
+		arg = strings.ReplaceAll(arg, str, fix)
+	}
+	arg = strings.ReplaceAll(arg, "{{FN%8}}", fmt.Sprint(index%8))
+	return arg
+}
+
 func splitArgs(str, templateDir string, index int) (args []string) {
 	for _, arg := range strings.Split(str, " ") {
 		if arg == "" {
 			continue
 		}
 		arg = strings.ReplaceAll(arg, "{{INDEX}}", fmt.Sprint(index))
 		arg = strings.ReplaceAll(arg, "{{TEMPLATE}}", templateDir)
+		if strings.Contains(arg, "vfio-pci,host=") && strings.Contains(arg, "{{FN%8}}") {
```

Comment:
@tarasmadan The FN check is still need otherwise the 1st if statement is entered if index > 7 whether FN is in the string or not. Moved the FN check inside handleVfioPciArg.
See: https://github.com/google/syzkaller/commit/c7220c09f5461bfed5769b38ce8b019b56525414